base_classes/NXhistory.nxdl.xml

```diff
@@ -0,0 +1,74 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+# For further information, see http://www.nexusformat.org
+-->
+<definition xmlns="http://definition.nexusformat.org/nxdl/3.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" category="base" type="group" name="NXhistory" extends="NXobject" xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd">
+    <doc>
+         A set of activities that occurred to a physical entity prior/during experiment.
+         
+         Ideally, a full report of the previous operations (or links to a chain of operations).
+         Alternatively, notes allow for additional descriptors in any format.
+    </doc>
+    <group type="NXactivity">
+        <doc>
+             Any activity that was performed on the physical entity prior or during the experiment. In
+             the future, if there is base class inheritance, this can describe any activity,
+             including processes and measurements.
+        </doc>
+    </group>
+    <!--For now, one workaround would be to have NXactivity as a application definition with a subentry.
+subentry(NXsuxbentry):
+doc: |
+Any activity that was performed on the physical entity prior or during the experiment.
+definition: ["NXactivity"]-->
+    <group type="NXphysical_process">
+        <doc>
+             Any physical process that was performed on the physical entity prior or during the
+             experiment.
+        </doc>
+    </group>
+    <group type="NXchemical_process">
+        <doc>
+             Any chemical process that was performed on the physical entity prior or during the
+             experiment.
+        </doc>
+    </group>
+    <group type="NXidentifier">
+        <doc>
+             An ID or reference to the location or a unique (globally
+             persistent) identifier of e.g. another file which gives
+             as many as possible details of the history event.
+        </doc>
+    </group>
+    <!--There should be more activities here, like measurement.-->
+    <group name="notes" type="NXnote">
+        <doc>
+             A descriptor to keep track of the treatment of the physical entity before or during the
+             experiment (NXnote allows to add pictures, audio, movies). Alternatively, a
+             reference to the location or a unique identifier or other metadata file. In the
+             case these are not available, free-text description.
+             This should only be used in case that there is no rigorous description
+             using the base classes above. This field can also be used to pull in any activities
+             that are not well described by an existing base class definition.
```

Comment:
Add comment that any number of NXnotes are allowed for extra details.

Reply:
done